sources/packages/backend/apps/api/src/route-controllers/config/_tests_/e2e/config.controller.getConfig.e2e-spec.ts

```diff
@@ -0,0 +1,76 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { Test, TestingModule } from "@nestjs/testing";
+import { ConfigModule } from "@sims/utilities/config";
+import { setGlobalPipes } from "../../../../utilities";
+import { CreateTestingModuleResult } from "../../../../testHelpers";
+import * as request from "supertest";
+import { ConfigController } from "../../config.controller";
+
+describe("ConfigController(e2e)-getConfig", () => {
+  let app: INestApplication;
+  const originalEnv = process.env;
+  const fakeEnvVariables = {
+    KEYCLOAK_AUTH_URL: "https://keycloak-fake-url",
+    KEYCLOAK_REALM: "keycloak_fake_realm",
+    KEYCLOAK_CLIENT_STUDENT: "student",
+    KEYCLOAK_CLIENT_INSTITUTION: "institution",
+    KEYCLOAK_CLIENT_AEST: "aest",
+    KEYCLOAK_CLIENT_SUPPORTING_USERS: "supporting-users",
+    SITE_MINDER_LOGOUT_URL: "https://fake-siteminder-logout-url",
+  };
+
+  beforeAll(async () => {
+    jest.resetModules();
+    process.env = {
+      ...originalEnv,
+      ...fakeEnvVariables,
+    };
+    const { nestApplication } = await createTestingConfigModule();
+    app = nestApplication;
+  });
+
+  it("Should return config values", async () => {
+    // Arrange Act/Assert
+    await request(app.getHttpServer())
+      .get("/config")
+      .expect(HttpStatus.OK)
+      .expect({
+        auth: {
+          url: fakeEnvVariables.KEYCLOAK_AUTH_URL,
+          realm: fakeEnvVariables.KEYCLOAK_REALM,
+          clientIds: {
+            student: fakeEnvVariables.KEYCLOAK_CLIENT_STUDENT,
+            institution: fakeEnvVariables.KEYCLOAK_CLIENT_INSTITUTION,
+            aest: fakeEnvVariables.KEYCLOAK_CLIENT_AEST,
+            supportingUsers: fakeEnvVariables.KEYCLOAK_CLIENT_SUPPORTING_USERS,
+          },
+          externalSiteMinderLogoutUrl: fakeEnvVariables.SITE_MINDER_LOGOUT_URL,
+        },
+      });
+  });
+
+  afterAll(async () => {
+    await app?.close();
+  });
+});
+
+/**
+ * API root module only with config module and config controller.
+ ** This module allows to mock any environment variable including
+ ** keycloak environment variables.
+ ** This module is exclusively for Config e2e tests.
+ * @returns test config module as root application module.
+ */
+export async function createTestingConfigModule(): Promise<CreateTestingModuleResult> {
```

Comment:
I believe that there is no need to have a method exported in the spec file.
IMO, the below code could just be moved to replace the code at line 28.
```ts
 const module: TestingModule = await Test.createTestingModule({
    imports: [ConfigModule],
    controllers: [ConfigController],
  }).compile();
```

Reply:
But what about the http application required to run http server.